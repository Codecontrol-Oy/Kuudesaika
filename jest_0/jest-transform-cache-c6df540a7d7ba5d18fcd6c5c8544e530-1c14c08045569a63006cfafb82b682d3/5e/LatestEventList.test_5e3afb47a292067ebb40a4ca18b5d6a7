'use strict';





jest.mock('react-router', function () {return { browserHistory: { push: jest.fn() } };});var _react = require('react');var _react2 = _interopRequireDefault(_react);var _LatestEventList = require('./LatestEventList');var _LatestEventList2 = _interopRequireDefault(_LatestEventList);var _Organizations = require('../../../mock/Organizations.json');var _Organizations2 = _interopRequireDefault(_Organizations);var _Events = require('../../../mock/Events.json');var _Events2 = _interopRequireDefault(_Events);var _sinon = require('sinon');var _sinon2 = _interopRequireDefault(_sinon);var _reactRouter = require('react-router');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe("<LatestEventList />", function () {

    it("Matches snapshot", function () {
        var latestEventList = shallow(
        _react2.default.createElement(_LatestEventList2.default, { organizations: _Organizations2.default.organizations, latestEvents: _Events2.default.events }));

        expect(latestEventList).toMatchSnapshot();
    });

    it("Should render LatestEvent components", function () {
        var latestEventList = shallow(
        _react2.default.createElement(_LatestEventList2.default, { organizations: _Organizations2.default.organizations, latestEvents: _Events2.default.events }));

        expect(latestEventList.find('LatestEvent').length).toEqual(20);
    });
});