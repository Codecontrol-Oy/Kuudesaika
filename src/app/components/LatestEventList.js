import React from 'react';
import PropTypes from 'prop-types';
import LatestEvent from './LatestEvent.js';
export default class LatestEventList extends React.Component {
  constructor (props) {
    super(props);
    self = this;
  }

  getOrganizationName = (id) => {
      for (let i = 0; i < this.props.organizations.length; i++) {
        if (this.props.organizations[i].id === id) {
            return this.props.organizations[i].name;
        }
      }
  }

    getOrganization = (id) => {
      for (let i = 0; i < self.props.organizations.length; i++) {
        if (self.props.organizations[i].id.toString() === id) {
            return self.props.organizations[i];
        }
      }
  }

  render () {
    let latest_events = [];
    if (this.props.latestEvents) {
        this.props.latestEvents.sort((a, b) => b.origin_id - a.origin_id).map(function (event) {
            const splitArray = event.organization.split('/');
            const organization_id = splitArray[splitArray.length - 2];
            const organization = self.getOrganization(organization_id);
            latest_events.push(<LatestEvent key={event.id + '_latest_event'} organization={organization} start_date={event.start_date} id={event.id} />);
        });
    }
    return (
        <div>
            {latest_events}
        </div>
    );
  }
}

LatestEventList.propTypes = {
    latestEvents: PropTypes.array,
    organizations: PropTypes.array
};
