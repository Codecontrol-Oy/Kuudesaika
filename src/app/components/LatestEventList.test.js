import React from 'react';
import LatestEventList from './LatestEventList';
import org_mock from '../../../mock/Organizations.json';
import event_mock from '../../../mock/Events.json';
import sinon from 'sinon';
import { browserHistory } from 'react-router';
jest.mock('react-router', ()=>({browserHistory: {push: jest.fn()}}))

describe("<LatestEventList />", () => {

    it("Matches snapshot", () => {
        const latestEventList = shallow(
            <LatestEventList organizations={org_mock.organizations} latestEvents={event_mock.events}  />
        );
        expect(latestEventList).toMatchSnapshot();
    });

    it("Should render LatestEvent components", () => {
        const latestEventList = shallow(
            <LatestEventList organizations={org_mock.organizations} latestEvents={event_mock.events}  />
        );
        expect(latestEventList.find('LatestEvent').length).toEqual(20);
    });
}) 