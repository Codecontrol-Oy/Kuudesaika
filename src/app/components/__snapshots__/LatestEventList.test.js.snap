// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LatestEventList /> Matches snapshot 1`] = `
<div>
  <LatestEvent
    id={1}
    start_date="2013-04-18"
  />
  <LatestEvent
    id={20}
    start_date="2012-03-07"
  />
  <LatestEvent
    id={19}
    start_date="2012-02-08"
  />
  <LatestEvent
    id={18}
    start_date="2012-01-25"
  />
  <LatestEvent
    id={17}
    start_date="2013-06-12"
  />
  <LatestEvent
    id={16}
    start_date="2013-05-23"
  />
  <LatestEvent
    id={15}
    start_date="2013-04-25"
  />
  <LatestEvent
    id={14}
    start_date="2013-04-03"
  />
  <LatestEvent
    id={13}
    start_date="2013-03-14"
  />
  <LatestEvent
    id={12}
    start_date="2013-02-05"
  />
  <LatestEvent
    id={11}
    start_date="2012-12-20"
  />
  <LatestEvent
    id={10}
    start_date="2012-11-22"
  />
  <LatestEvent
    id={9}
    start_date="2012-11-08"
  />
  <LatestEvent
    id={8}
    start_date="2012-10-25"
  />
  <LatestEvent
    id={7}
    start_date="2012-10-11"
  />
  <LatestEvent
    id={6}
    start_date="2012-09-27"
  />
  <LatestEvent
    id={5}
    start_date="2012-08-16"
  />
  <LatestEvent
    id={4}
    start_date="2012-06-07"
  />
  <LatestEvent
    id={3}
    start_date="2012-05-24"
  />
  <LatestEvent
    id={2}
    start_date="2012-04-26"
  />
</div>
`;
